.github/workflows/ci.yml

```diff
@@ -12,16 +12,20 @@ jobs:
   build:
     runs-on: ubuntu-latest
 
+    # Test recent LTS versions of Node.js, as well as the exact version from `node-version-file`.
     strategy:
       matrix:
-        node-version: [12.x, 14.x]
+        node-version: [18, 20, 22, ""]
 
     steps:
-      - uses: actions/checkout@v2
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 #v4.2.1
+      - uses: actions/setup-node@0a44ba7841725637a19e28fa30b79a866c81b0a6 #v4.0.4
+        name: Use Node.js ${{ matrix.node-version }}
         with:
+          cache: npm
           node-version: ${{ matrix.node-version }}
-      - run: npm install
+          node-version-file: .nvmrc
+      - run: npm ci
```

Comment:
This works.

If you want to disable the postinstall scripts, `npm ci --ignore-scripts` would be the option to set.
This might prevent malicious scripts from being executed.